pkg/agent/baker.go

```diff
@@ -260,10 +260,16 @@ func normalizeResourceGroupNameForLabel(resourceGroupName string) string {
 func validateAndSetLinuxNodeBootstrappingConfiguration(config *datamodel.NodeBootstrappingConfiguration) {
 	// If using kubelet config file, disable DynamicKubeletConfig feature gate and remove dynamic-config-dir
 	// we should only allow users to configure from API (20201101 and later)
+	profile := config.AgentPoolProfile
 	if config.KubeletConfig != nil {
 		kubeletFlags := config.KubeletConfig
 		delete(kubeletFlags, "--dynamic-config-dir")
-		delete(kubeletFlags, "--image-pull-progress-deadline")
+		delete(kubeletFlags, "--non-masquerade-cidr")
```

Comment:
Does 1.24 not support this flag --non-masquerade-cidr? Should we also remove this for Windows?

Reply:
Yes, k8s 1.24 does not support --non-masquerade-cidr flag

Reply:
It was removed in this PR: https://github.com/kubernetes/kubernetes/pull/107096

Reply:
It does seem like it can be removed for Windows too